### ISSUE TITLE
image decoding fails when connecting via the proxy

### ISSUE THREAD

**Issue by totaam** [Xpra-org/xpra-html5#322]
`failed to create image bitmap from webp [object Bl…tateError: The source image could not be decoded.`

Not sure when this started. Very odd.
I even copied the `Uint8Array` data from the Javascript console, then:
```js
const data = Uint8Array([82,73,70,70,190,28,0,0,87,69,66,80,86,80,56,76,177,28,0,0,47,248,193,78,0,39,...])
const blob = new Blob([bytes], { type: `image/webp` });
const p = createImageBitmap(blob);
```
```js
Promise {<fulfilled>: ImageBitmap}\
    [[Prototype]]: Promise[[PromiseState]]: "fulfilled"\
    [[PromiseResult]]: ImageBitmapheight: 316width: 505\
        [[Prototype]]: ImageBitmapclose: ƒ close()\
        height: (...)width: (...)\
        constructor: ƒ ImageBitmap()Symbol(Symbol.toStringTag): "ImageBitmap"get \
        height: ƒ height()get width: ƒ width()[[Prototype]]: Object
```

So the console can decode it, but the worker can't?
What is different about the proxy connection?

Pillow can also decode the data:
```python
from io import BytesIO
b = BytesIO(bytearray([82,73,70,70,190,28,0,0,87,69,66,80,86,80,56,76,177,28,0,0,47,248,193,78,0,39,...])
from PIL import Image
img = Image.open(b)
img
<PIL.WebPImagePlugin.WebPImageFile image mode=RGB size=505x316 at 0x7F779D3841A0>
img.size
(505, 316)
```

And the python client doesn't have any problems connecting via the proxy either.

**Comment by totaam** [comment]
Workaround in 614c1b538c98268564c60d18c2571ca868f9e70e
Decoding the same data works fine in the client, so we forward it there.

**Comment by totaam** [comment]
The spec [`createImageBitmap`](https://html.spec.whatwg.org/multipage/imagebitmap-and-animations.html#images-2) says: _Rejects the promise with an ["InvalidStateError"](https://webidl.spec.whatwg.org/#invalidstateerror) [DOMException](https://webidl.spec.whatwg.org/#dfn-DOMException) if the source image is not in a valid state (e.g., an [img](https://html.spec.whatwg.org/multipage/embedded-content.html#the-img-element) element that hasn't loaded successfully, an [ImageBitmap](https://html.spec.whatwg.org/multipage/imagebitmap-and-animations.html#imagebitmap) object whose [[[Detached]]](https://html.spec.whatwg.org/multipage/structured-data.html#detached) internal slot value is true, an [ImageData](https://html.spec.whatwg.org/multipage/canvas.html#imagedata) object whose [data](https://html.spec.whatwg.org/multipage/canvas.html#dom-imagedata-data) attribute value's [[ViewedArrayBuffer]] internal slot is detached, or a [Blob](https://w3c.github.io/FileAPI/#dfn-Blob) whose data cannot be interpreted as a bitmap image)._

But surely the blob data is correct if we can decode it in the client?

**Comment by totaam** [comment]
This has something to do with accessing the underlying [`buffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Typed_arrays#buffer) of the `Uint8Array`.
I really do not understand why this makes any difference, because when I dump the `img_data` on reception of `draw` packets - or even later in the decode worker, it is always a `Uint8Array`...

The (hopefully temporary) solution is to turn off zerocopy if we fail to decode.

I thought that this may have something to do with image data being inlined, or not, when being tagged by the proxy here:
https://github.com/Xpra-org/xpra/blob/04baa0522821c7f8d8474a352a27f98b4eda159b/xpra/server/proxy/instance_base.py#L373
But manually overriding the `can_inline` flag does not seem to make any difference.

**Comment by totaam** [comment]
Fixed in 105ba6a6a6906956d932044c78bbe26cf78836fc : the `rencodeplus` packet decoder uses a subarray:
https://github.com/Xpra-org/xpra-html5/blob/105ba6a6a6906956d932044c78bbe26cf78836fc/html5/js/lib/rencode.js#L292
As per [`TypedArray.subarray`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/subarray): _returns a new typed array on the same [ArrayBuffer](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) store_
So this causes problems later on when we try to transfer it to the decode worker - somehow.
An alternative fix would have been to use [`TypedArray.slice`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/slice).

The other workarounds are worth keeping anyway.

**Comment by totaam** [comment]
The simplest fix is available in [v16.1](https://github.com/Xpra-org/xpra-html5/releases/tag/v16.1)
